src/modules/Api.ts

```diff
@@ -183,6 +190,89 @@ export class Api {
     });
   }
 
+  public async canonicalOrder({
+    order: {
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration = new BigNumber(FOUR_WEEKS_IN_SECONDS),
+      limitFee,
+    },
+    fillOrKill,
+    postOnly,
+    clientId,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
+    order: {
+      side: ApiSide,
+      market: ApiMarketName,
+      amount: Integer | string,
+      price: string,
```

Comment:
create a new type `BigNumberable` in types that has type `BigNumber | string | number` and use that for amount, price, expiration, limitFee

Reply:
I'm still going to be making BigNumber's from these though since they only _might_ be BigNumbers when they are passed in?